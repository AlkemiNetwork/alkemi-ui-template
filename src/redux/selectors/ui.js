// Libs
import { matchPath } from "react-router";
import { createSelector } from "reselect";

// Constants
import { HOME } from "../../constants/routes";

// Item selection selector
const getPathName = (state, props) => state.router.location.pathname;
export const isItemSelected = createSelector(
    getPathName,
    pathname => !!matchPath(pathname, { path: HOME })
);
