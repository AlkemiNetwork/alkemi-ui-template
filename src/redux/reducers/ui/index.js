import { combineReducers } from "redux";

import sidebar from "./sidebar";

export default combineReducers({
    sidebar
    // other ui reducers go here
});
