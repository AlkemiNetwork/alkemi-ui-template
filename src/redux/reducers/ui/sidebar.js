import { OPEN_SIDEBAR, CLOSE_SIDEBAR } from "../../actions/ui";

const initialState = {
    open: false
};

export default (state = initialState, { type }) => {
    switch (type) {
        case OPEN_SIDEBAR:
            return {
                ...state,
                open: true
            };

        case CLOSE_SIDEBAR:
            return {
                ...state,
                open: false
            };

        default:
            return state;
    }
};
