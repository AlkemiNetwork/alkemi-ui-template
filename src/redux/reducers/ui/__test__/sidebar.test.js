import { OPEN_SIDEBAR, CLOSE_SIDEBAR } from "../../../actions/ui";
import sidebarReducer from "../sidebar";

describe("sidebarReducer", () => {
    let state = {};

    it("default state match", () => {
        expect(sidebarReducer(undefined, {})).toMatchSnapshot();
    });

    describe(`On ${OPEN_SIDEBAR}`, () => {
        beforeEach(() => {
            state = sidebarReducer({ open: false }, { type: OPEN_SIDEBAR });
        });

        it("open=true", () => {
            expect(state.open).toEqual(true);
        });

        it("state matches snapshot", () => {
            expect(state).toMatchSnapshot();
        });
    });

    describe(`On ${CLOSE_SIDEBAR}`, () => {
        beforeEach(() => {
            state = sidebarReducer({ open: true }, { type: CLOSE_SIDEBAR });
        });

        it("open=false", () => {
            expect(state.open).toEqual(false);
        });

        it("state matches snapshot", () => {
            expect(state).toMatchSnapshot();
        });
    });

    describe("On default", () => {
        beforeEach(() => {
            state = sidebarReducer(undefined, { type: undefined });
        });

        it("state matches snapshot", () => {
            expect(state).toMatchSnapshot();
        });
    });
});
