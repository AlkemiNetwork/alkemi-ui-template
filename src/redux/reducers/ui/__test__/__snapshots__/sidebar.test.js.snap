// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sidebarReducer On CLOSE_SIDEBAR state matches snapshot 1`] = `
Object {
  "open": false,
}
`;

exports[`sidebarReducer On OPEN_SIDEBAR state matches snapshot 1`] = `
Object {
  "open": true,
}
`;

exports[`sidebarReducer On default state matches snapshot 1`] = `
Object {
  "open": false,
}
`;

exports[`sidebarReducer default state match 1`] = `
Object {
  "open": false,
}
`;
